### PR TITLE
Remove NotQueryBuilder

### LINKED ISSUE #13761: Not Query: Investigate if we can deprecate short version
Currently the Not Query can have two version, either directly starting with the inner query or nesting it in an additinal `filter` element:

```
"not" : {
                "range" : {
                    "postDate" : {
                        "from" : "2010-03-01",
                        "to" : "2010-04-01"
                    }
                }
          }
```
or 
```
"not" : {
                "filter" :  {
                    "range" : {
                        "postDate" : {
                            "from" : "2010-03-01",
                            "to" : "2010-04-01"
                        }
                    }
                }
          }
```

The current NotQueryParser supports both versions, but it would be beneficial to be able to remove one version, so this issue is here to discuss whether and how we can deprecate the short version.

### PR DESCRIPTION
**Pull request by cbuescher**
The NotQueryBuilder should be deprecated on the 2.x branches
and can be removed with the next major version. It's equivalent is
the booean query mustNot() clause.

Closes #13761

### CODE REVIEW

**Comment by javanna** [comment]
LGTM can you add a note to the migration guide please? Also open an issue to the [migration project](https://github.com/elastic/elasticsearch-migration) so we don't forget about it?

**Comment by clintongormley** [comment]
The migration plugin doesn't look at queries, so no need to open an issue there

**Comment by cbuescher** [comment]
@javanna added entry to the `migrate_3_0.asciidoc`, or do you think this better belongs with the deprecation into the corresponding `migrate_2_1.asciidoc`?

File: docs/reference/migration/migrate_3_0.asciidoc [hunk 1]
```diff
@@ -285,6 +285,11 @@ The two individual setters for existence() and nullValue() were removed in favou
 optional constructor settings in order to better capture and validate their interdependent
 settings at construction time.
 
+==== NotQueryBuilder
+
+The NotQueryBuilder which was deprecated in 2.1.0 is removed. As a replacement use BooleanQueryBuilder
```
Comment (javanna) [inline docs/reference/migration/migrate_3_0.asciidoc]: BoolQueryBuilder ? ;) can you include an example of what we used to have and what you have after the migration?

**Comment by javanna** [comment]
migrate_3_0.asciidoc is fine, thanks! I left a comment.

File: docs/reference/migration/migrate_3_0.asciidoc (hunk 1 above)
Comment (cbuescher) [inline docs/reference/migration/migrate_3_0.asciidoc]: Sure, added very brief example of the change in the API.

File: docs/reference/migration/migrate_3_0.asciidoc [hunk 2]
```diff
@@ -285,6 +285,12 @@ The two individual setters for existence() and nullValue() were removed in favou
 optional constructor settings in order to better capture and validate their interdependent
 settings at construction time.
 
+==== NotQueryBuilder
+
+The NotQueryBuilder which was deprecated in 2.1.0 is removed. As a replacement use BoolQueryBuilder
+with added mustNot() clause. So instead of using `NotQueryBuilder(filter)` now use 
+`BoolQueryBuilder().mustNot(filter)`.
```
Comment (javanna) [inline docs/reference/migration/migrate_3_0.asciidoc]: sorry for being picky, can you prepend the snippets this `new`? They are supposed to call constructors no? That's it ;)

Comment (cbuescher) [inline docs/reference/migration/migrate_3_0.asciidoc]: No problem. Also rebased and squashed while I'm at it.

**Comment by javanna** [comment]
LGTM

**Pull request merged**